### PR TITLE
dz60rgb_wkl move features, rgb_matrix to info.json #21967 

### PR DESCRIPTION
**Pull request by haunt98**
dz60rgb_wkl move features, rgb_matrix from rules.mk, config.h to info.json thanks to data driven config.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by haunt98** [comment]
Hi @dunk2k, I close last PR https://github.com/qmk/qmk_firmware/pull/21967 to re open here because I accidently rename my branch in my fork.

I have fixed the lint issue by adding copyright header as you suggest.

**Comment by dunk2k** [comment]
> I have fixed the lint issue by adding copyright header as you suggest.

Great stuff, thank you 🙂

**Comment by haunt98** [comment]
Hi @moyi4681, please take a look if you have spare time. Because readme suggest that you are maintainer of dz60rgb_wkl PCB. You can ignore this if you want, thank you.

File: keyboards/dztech/dz60rgb_wkl/v2/info.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
     "usb": {
         "pid": "0x1321",
-        "device_version": "2.0.0"
+        "device_version": "2.0.0",
+        "no_startup_check": false
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/v2/info.json:5]:
```suggestion
        "device_version": "2.0.0"
```

File: keyboards/dztech/dz60rgb_wkl/v2_1/config.h [hunk 2]
```diff
@@ -1,75 +1,15 @@
...
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright 2021 trankimtung
+// Copyright 2023 haunt98 <hauvipapro@gmail.com>
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/v2_1/config.h:2]:
```suggestion
```

File: keyboards/dztech/dz60rgb_wkl/v2/config.h [hunk 3]
```diff
@@ -1,54 +1,12 @@
+// Copyright 2023 haunt98 <hauvipapro@gmail.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/v2/config.h:3]:
```suggestion
```

File: keyboards/dztech/dz60rgb_wkl/v1/info.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
 {
     "usb": {
         "pid": "0x1320",
-        "device_version": "1.0.0"
+        "device_version": "1.0.0",
+        "no_startup_check": false
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/v1/info.json:5]:
```suggestion
        "device_version": "1.0.0"
```

File: keyboards/dztech/dz60rgb_wkl/v1/config.h [hunk 5]
```diff
@@ -1,58 +1,13 @@
+// Copyright 2023 haunt98 <hauvipapro@gmail.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/v1/config.h:3]:
```suggestion
```

File: keyboards/dztech/dz60rgb_wkl/keymaps/via/keymap.c [hunk 6]
```diff
@@ -1,5 +1,9 @@
+// Copyright 2023 haunt98 <hauvipapro@gmail.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/keymaps/via/keymap.c:3]:
```suggestion
```

File: keyboards/dztech/dz60rgb_wkl/keymaps/default/keymap.c [hunk 7]
```diff
@@ -1,5 +1,9 @@
+// Copyright 2023 haunt98 <hauvipapro@gmail.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/keymaps/default/keymap.c:3]:
```suggestion
```

File: keyboards/dztech/dz60rgb_wkl/info.json [hunk 8]
```diff
@@ -6,6 +6,53 @@
...
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "backlight": false,
+        "rgblight": false,
+        "audio": false,
```
Comment (zvecr) [inline keyboards/dztech/dz60rgb_wkl/info.json:21]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
* Per the PR checklist `keyboard updates and refactors (eg. to data driven) must go through develop`
* Lint failures for licence headers wont stop your PR from merging
  * You would not be able to claim copyright with your changes anyway
* Adding clang-format comments to keymaps provides little value as they are currently not required to be formatted

**Comment by haunt98** [comment]
Hi @zvecr, I have updated following your suggestion, also remove redundant clang-format, thank you.

**Pull request closed without merging**